src/core/Form/SearchInput/SearchInput.tsx

```diff
@@ -228,43 +231,57 @@ class BaseSearchInput extends Component<SearchInputProps> {
         <LabelText htmlFor={this.id} labelMode={labelMode} as="label">
           {labelText}
         </LabelText>
-        <HtmlDiv className={searchInputClassNames.functionalityContainer}>
-          <HtmlDiv className={searchInputClassNames.inputElementContainer}>
-            <HtmlInputWithRef
-              {...passProps}
-              {...getDescribedBy()}
-              forwardRef={this.inputRef}
-              aria-invalid={status === 'error'}
-              id={this.id}
-              className={searchInputClassNames.inputElement}
-              type="search"
-              role="searchbox"
-              value={this.state.value}
-              placeholder={visualPlaceholder}
-              onChange={(event: ChangeEvent<HTMLInputElement>) => {
-                conditionalSetState(event.currentTarget.value);
-              }}
-              onKeyPress={onKeyPress}
-              onKeyDown={onKeyDown}
-            />
-          </HtmlDiv>
-          <HtmlButton {...clearButtonProps}>
-            <VisuallyHidden>{clearButtonLabel}</VisuallyHidden>
-            <Icon
-              aria-hidden={true}
-              icon="close"
-              className={searchInputClassNames.clearIcon}
-            />
-          </HtmlButton>
-          <HtmlButton {...searchButtonDerivedProps}>
-            <VisuallyHidden>{searchButtonLabel}</VisuallyHidden>
-            <Icon
-              aria-hidden={true}
-              icon="search"
-              className={searchInputClassNames.searchIcon}
-            />
-          </HtmlButton>
-        </HtmlDiv>
+        <Debounce waitFor={this.props.debounce || 0}>
+          {(debouncer: Function, cancelDebounce: Function) => (
+            <HtmlDiv className={searchInputClassNames.functionalityContainer}>
+              <HtmlDiv className={searchInputClassNames.inputElementContainer}>
+                <HtmlInputWithRef
+                  {...passProps}
+                  {...getDescribedBy()}
+                  forwardRef={this.inputRef}
+                  aria-invalid={status === 'error'}
+                  id={this.id}
+                  className={searchInputClassNames.inputElement}
+                  type="search"
+                  role="searchbox"
+                  value={this.state.value}
+                  placeholder={visualPlaceholder}
+                  onChange={(event: ChangeEvent<HTMLInputElement>) => {
+                    const eventValue = event.currentTarget.value;
+                    conditionalSetState(eventValue);
+                    if (propOnChange) {
+                      debouncer(propOnChange, eventValue);
+                    }
+                  }}
+                  onKeyPress={onKeyPress}
+                  onKeyDown={onKeyDown}
+                />
+              </HtmlDiv>
+              <HtmlButton
+                {...clearButtonProps}
+                onClick={() => {
+                  onClear();
+                  cancelDebounce();
```

Comment:
Now when user presses clear button the debounce is then cancelled and the `onChange` is called immediately even after the user continues to write. So possibly doing some unnecessary logic to search from API etc if it is not taken care by the user.

Maybe we should tell this in documentation to library user?

Reply:
As discussed separately the consensus was that let's leave this as it is now.